common/types/hash.go

```diff
@@ -0,0 +1,35 @@
+package types
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+)
+
+// Hash represents a 32-byte hash
+type Hash = ethcommon.Hash
```

Comment:
Using assignment above, we get all the JSON handing, as well as the `String()` and `Hex()` methods of the original type. Only downside is that we can't extend the type with our own methods like `IsZero`.  I changed that method to `IsHashZero` below and it just takes the hash as a parameter.